### PR TITLE
OCPBUGS-42610: filter out pod events during vsphere snapshot options tests

### PR DESCRIPTION
**Pull request by RomanBednar**

### CODE REVIEW

**Comment by RomanBednar** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-vsphere-ovn-serial

**Comment by dobsonj** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-vsphere-ovn-serial

**Comment by RomanBednar** [comment]
/payload-job periodic-ci-openshift-release-master-nightly-4.18-e2e-vsphere-ovn-serial

File: pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go [hunk 1]
```diff
@@ -433,6 +433,19 @@ func NewUniversalPathologicalEventMatchers(kubeConfig *rest.Config, finalInterva
...
+	// Repeating events about pod creation are expected for snapshot options tests in vsphere csi driver.
+	// The tests change clusterCSIDriver object and have to rollout new pods to load new configuration.
+	registry.AddPathologicalEventMatcherOrDie(&SimplePathologicalEventMatcher{
+		name: "VsphereConfigurationTestsRollOutTooOften",
+		locatorKeyRegexes: map[monitorapi.LocatorKey]*regexp.Regexp{
+			monitorapi.LocatorNamespaceKey: regexp.MustCompile(`^openshift-cluster-csi-drivers$`),
+			monitorapi.LocatorE2ETestKey:   regexp.MustCompile(`.*snapshot options in clusterCSIDriver.*`),
+		},
+		messageReasonRegex: regexp.MustCompile(`(^SuccessfulCreate$|^SuccessfulDelete$)`),
+		messageHumanRegex:  regexp.MustCompile(`^Created pod.*vmware-vsphere-csi-driver.*`),
```
Comment (xueqzhan) [inline pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go:445]: In the delete case, should we match "Deleted pod: vmware-vsphere-csi-driver-node-zhcsx" too?

**Comment by xueqzhan** [comment]
/lgtm

Please make sure https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-release-master-nightly-4.18-e2e-vsphere-ovn-upi-serial/1847703900736458752 is covered.

File: pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go (hunk 1 above)
Comment (RomanBednar) [reply pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go]: Probably yes, added.

**Pull request merged**